src/tools/extractor/version.c

```
@@ -35,7 +35,92 @@ void CheckGameVersion(const char *path) {
+    }
+
+    snprintf(fcheck, sizeof(fcheck), "%s/SLES_010.95", path);
+    if (plFileExists(fcheck)) {
+      LogWarn("Failed to find PS-X EXE, unable to determine region!\n");
+      //version_info.platform = PLATFORM_PSX_PAL?;
+      version_info.region = REGION_ENG;
+    }
+
+    snprintf(fcheck, sizeof(fcheck), "%s/SLES_027.66", path);
+    if (plFileExists(fcheck)) {
+      LogWarn("Failed to find PS-X EXE, unable to determine region!\n");
+      //version_info.platform = PLATFORM_PSX_PAL?;
+      version_info.region = REGION_FRE;
+    }
+
+    snprintf(fcheck, sizeof(fcheck), "%s/SLES_027.67", path);
+    if (plFileExists(fcheck)) {
+      LogWarn("Failed to find PS-X EXE, unable to determine region!\n");
+      //version_info.platform = PLATFORM_PSX_PAL?;
+      version_info.region = REGION_GER;
+    }
+
+    snprintf(fcheck, sizeof(fcheck), "%s/SLES_027.68", path);
+    if (plFileExists(fcheck)) {
+      LogWarn("Failed to find PS-X EXE, unable to determine region!\n");
+      //version_info.platform = PLATFORM_PSX_PAL?;
+      version_info.region = REGION_SPA;
+    }
+
+    snprintf(fcheck, sizeof(fcheck), "%s/SLES_027.69", path);
+    if (plFileExists(fcheck)) {
+      LogWarn("Failed to find PS-X EXE, unable to determine region!\n");
+      //version_info.platform = PLATFORM_PSX_PAL?;
+      version_info.region = REGION_ITA;
+    }
+
+#if false
+    // TODO: Alternative approach reading the language.lng file.    
+    // Language ids:
+    // 0 = ENG_GB
+    // 1 = ENG_US
+    // 2 = FRE (no proof)
+    // 3 = GER
+    // 4 = SPA (no proof)
+    // 5 = ITA (no proof)
+
+    snprintf(fcheck, sizeof(fcheck), "%s/language.lng", path);
+    if (!plFileExists(fcheck)) {
+      LogWarn("Failed to find language.lng, unable to determine language!\n");
+      return;
+    }
+
+    PLFile *lgn_file = plOpenFile(path, false);
+    if (lgn_file == NULL) {
+      LogWarn("Failed to load \"%s\", aborting!\n", path);
+      return NULL;
```

Should throw an Error and abort (just change LogWarn to Error)

Before ! add '(%s)' and use plGetError